frame/alliance/src/lib.rs

```diff
@@ -306,6 +314,10 @@ pub mod pallet {
 
 		/// Weight information for extrinsics in this pallet.
 		type WeightInfo: WeightInfo;
+
+		/// The period required to pass from retirement notice for a member to retire.
+		/// Supposed to be greater than time required to `kick_member`.
+		type RetirementPeriod: Get<Self::BlockNumber>;
```

Comment:
@muharem 
I am not sure about `Self:BlockNumber` type here, 
for me, `Get<Self::BlockNumber>` evokes, that something happens at that BlockNumber,

but bellow you just add it `frame_system::Pallet::<T>::block_number() + T::RetirementPeriod::get(),`

I guess, that `Get<Self::BlockNumber>` is more like `offset`, but I dont know, maybe it is ok.




just one thing: `block_number() + T::RetirementPeriod::get()` what about `saturating_add` instead?

Reply:
I have updated the doc and using `saturating_add` now

Reply:
ok, probably there is some consensus that "number of blocks" could be reperesented as BlockNumber
```
   pub const MINUTES: BlockNumber = 60_000 / (MILLISECS_PER_BLOCK as BlockNumber);
	pub const HOURS: BlockNumber = MINUTES * 60;
	pub const DAYS: BlockNumber = HOURS * 24;
	pub const WEEKS: BlockNumber = DAYS * 7;
```

Reply:
Yes, its very common, search for `Get<Self::BlockNumber>` within `substrate/frame`